src/extensions/default/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/PackageAnalyzerResultProvider.cs

```diff
@@ -90,22 +93,24 @@ private HashSet<AnalyzeResult> ExtractAnalysisResult(string fileLocation, IDepen
             {
                 results.Add(new()
                 {
-                    RuleId = Id,
-                    RuleName = this.Name,
+                    RuleId = RuleId,
+                    RuleName = RuleName,
+                    HelpUri = _helpUri,
+                    FullDescription = FullDescription,
                     FileLocation = fileLocation,
-                    ResultMessage = "No package updates needed",
+                    ResultMessage = "No package updates needed.",
                 });
             }
             else
             {
-                GetResults("Reference to ", " needs to be deleted ", analysisState.References.Deletions);
-                GetResults("Reference to ", " needs to be added ", analysisState.References.Additions);
-                GetResults("Package ", " needs to be deleted ", analysisState.Packages.Deletions);
-                GetResults("Package ", " needs to be added ", analysisState.Packages.Additions);
-                GetResults("Framework Reference to ", " needs to be deleted ", analysisState.FrameworkReferences.Deletions);
-                GetResults("Framework Reference to ", " needs to be added ", analysisState.FrameworkReferences.Additions);
-
-                void GetResults<T>(string name, string action, IReadOnlyCollection<Operation<T>> collection)
+                GetResults("UA103", "ReferenceShouldBeDeleted", "Reference to ", " needs to be deleted.", analysisState.References.Deletions);
+                GetResults("UA104", "ReferenceShouldBeAdded", "Reference to ", " needs to be added.", analysisState.References.Additions);
+                GetResults("UA105", "PackageShouldBeDeleted", "Package ", " needs to be deleted.", analysisState.Packages.Deletions);
+                GetResults("UA106", "PackageShouldBeAdded", "Package ", " needs to be added.", analysisState.Packages.Additions);
+                GetResults("UA107", "FrameworkShouldBeDeleted", "Framework Reference to ", " needs to be deleted.", analysisState.FrameworkReferences.Deletions);
+                GetResults("UA108", "FrameworkShouldBeAdded", "Framework Reference to ", " needs to be added.", analysisState.FrameworkReferences.Additions);
```

Comment:
This should be "FrameworkReferenceToBeAdded" and "FrameworkReferencesToBeDeleted" and for the rest to lets make it "ToBe" instead of "ShouldBe". Other than that LGTM

Reply:
changed!